src/QMCWaveFunctions/WaveFunctionComponent.h

```diff
@@ -170,6 +170,11 @@ class WaveFunctionComponent : public QMCTraits
   /** print the state, e.g., optimizables */
   virtual void reportStatus(std::ostream& os) = 0;
 
+  /** Register the component with the TWFPrototype wrapper.  Pure virtual, so will throw
+   * an error at compile time
+   */
+  virtual void register_TWF_Prototype(ParticleSet& P, TWFPrototype& twf);
```

Comment:
If register_TWF_Prototype is to query existing data form ParticleSet and TWF. It is better to have P const and this function const. Maybe const is not eligible on the function if register_TWF_Prototype does initialize internal data structures. I meant use const whenever feasible.

Reply:
Done.